geo/src/main/java/org/apache/carbondata/geo/GeoConstants.java

```diff
@@ -26,4 +26,34 @@ private GeoConstants() {
 
   // GeoHash type Spatial Index
   public static final String GEOHASH = "geohash";
+
+  // Regular expression to parse input polygons for IN_POLYGON_LIST
+  public static final String POLYGON_REG_EXPRESSION = "(?<=POLYGON \\(\\()(.*?)(?=(\\)\\)))";
```

Comment:
1. why 2 `((` and `))` for each POLYGON ((x1 y1, x2 y2, x3 y3, x1 y1)) ? 
2. Forcing user to give a space after POLYGON keyword. Otherwise match will not be successful. Shall we ignore these space restriction?

Reply:
we want to use the generic and fix input format,  search some others defination and decide it.
https://postgis.net/docs/ST_GeomFromText.html
https://stackoverflow.com/questions/8576228/geomfromtext-function-documentation
https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry#Geometric_objects